csrc/kernel_cache.h

```diff
@@ -75,19 +75,11 @@ class TORCH_CUDA_CU_API FusionKernelRuntime {
         });
   }
 
-  //! starts compilation async
-  void startAsyncCompile(const KernelArgumentHolder& inputs);
-
-  //! maps entries in `args` to fusion inputs.
-  //! Note that this function also pushes extra bits like dimension extent into
-  //! `args` for expression evaluator binding. So consider your `args` polluted
-  //! after this function and use it with caution.
-  void mapFusionInputsToArgs(
-      std::unordered_map<Val*, const ArgAbstract*>& tensor_map,
-      KernelArgumentHolder& args);
-
   //! Unified interface to run the managed kernels with given input
-  std::vector<at::Tensor> runWithInput(KernelArgumentHolder& args);
+  std::vector<at::Tensor> runWithInputs(KernelArgumentHolder&& args);
```

Comment:
I wanted to move the `args` with `runWithInputs` since it isn't used afterwards in the function.
```cpp
auto outputs = kernel_runtime->runWithInputs(std::move(args));
// csrc/kernel_cache.cpp:213
```

Reply:
I'm not sure if this is a good idea. There are already two types of interfaces, one taking `const KernelArgumentHolder&` and another `KernelArgumentHolder&`. I see that the `&&` version would be more efficient, but I don't think it's not that different from non-const reference. `args` being not used after the call doesn't seem to be a big problem. IIRC, there's no error checking even if you used moved objects. I think we should stick to a simpler and more consistent interface.